### ISSUE TITLE
How to generate  Roslyn analyzer findings report through command line

### ISSUE THREAD

**Issue by prasadmamidi7** [dotnet/roslyn-analyzers#3633]
Hi

Is there a way we can output the Roslyn analyzer code findings to an XML out put file e.g. similar to Visual studio error list window when we run code analysis. preferably using Directory.Build.Props or Directory.Build.Targets  file.

Thanks
Prasad Mamidi.

**Comment by mavasani** [comment]
You can generate an error log fine in SARIF format for all diagnostics. Specify the following MSBuild property in your MSBuild project file:

```xml
<PropertyGroup>
  <ErrorLog>%full_path_to_log_file%</ErrorLog>
</PropertyGroup>
```

If you want to share this property across multiple projects by putting it in Directory.Build.Props or targets, then your full path to log file should be unique per project, i.e. likely include either of the following `$(MSBuildProjectDirectory)` or `$(MSBuildProjectName)`.

**Comment by prasadmamidi7** [comment]
thank you. I am able to gernerated the report now.

**Comment by alexlee820** [comment]
I would like to ask how do you generate the report in command line ,could you give some step since I would like to generate it too

**Comment by mavasani** [comment]
@alexlee820 You can specify a command line switch “/errorlog:filepath” on csc/vbc command line to generate an error log file.
